lib/uring/uring.ml

```diff
@@ -344,19 +344,17 @@ let exit t =
 
 let with_id_full : type a. a t -> (Heap.ptr -> bool) -> a -> extra_data:'b -> a job option =
  fun t fn datum ~extra_data ->
-  match Heap.alloc t.data datum ~extra_data with
-  | exception Heap.No_space ->
-    check t;    (* Check if it's because we exited already. *)
-    None
-  | entry ->
-    let ptr = Heap.ptr entry in
-    let has_space = fn ptr in
-    if has_space then (
-      Some entry
-    ) else (
-      ignore (Heap.free t.data ptr : a);
-      None
-    )
+ let entry = try Heap.alloc t.data datum ~extra_data
+   with exc -> check t; raise exc
+ in
```

Comment:
Does this code really need to be reformatted? It seemed OK as it was (the new code is only shorter because the comment got deleted).
```suggestion
   match Heap.alloc t.data datum ~extra_data with
  | exception (Invalid_argument _ as ex) -> check t; raise ex
  | entry ->
```

Reply:
I was shuffling things around and this snuck in.